compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -2747,12 +2773,17 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 [trait_info] if trait_info.def_id.is_local() => {
                     err.span_note(
                         self.tcx.def_span(trait_info.def_id),
-                        format!(
-                            "`{}` defines an item `{}`, perhaps you need to {} it",
-                            self.tcx.def_path_str(trait_info.def_id),
-                            item_name,
-                            action
-                        ),
+                        fluent::hir_typeck_candidate_trait_note,
+                    )
+                    .set_arg("trait_name", self.tcx.def_path_str(trait_info.def_id))
+                    .set_arg("item_name", item_name)
+                    .set_arg(
+                        "postfix",
+                        if trait_missing_method {
```

Comment:
Please use diagnostic structs in `src/errors.rs` See [this](https://rustc-dev-guide.rust-lang.org/diagnostics/diagnostic-structs.html).